service/builder/exporters_builder.go

```diff
@@ -314,46 +314,35 @@ func exporterTypeMismatchErr(
 
 // createTraceProcessor creates a trace exporter based on provided factory type.
 func createTraceExporter(
-	factoryBase component.ExporterFactoryBase,
+	factory component.ExporterFactory,
 	logger *zap.Logger,
 	cfg configmodels.Exporter,
-) (component.TraceExporterBase, error) {
-	if factory, ok := factoryBase.(component.ExporterFactory); ok {
-		creationParams := component.ExporterCreateParams{Logger: logger}
-		ctx := context.Background()
-
-		// If exporter is of the new type (can manipulate on internal data structure),
-		// use ExporterFactory.CreateTraceExporter.
-		return factory.CreateTraceExporter(ctx, creationParams, cfg)
-	}
+) (component.TraceExporter, error) {
+	creationParams := component.ExporterCreateParams{Logger: logger}
+	ctx := context.Background()
 
-	// If exporter is of the old type (can manipulate on OC traces only),
-	// use ExporterFactoryOld.CreateTraceExporter.
-	return factoryBase.(component.ExporterFactoryOld).CreateTraceExporter(logger, cfg)
+	// If exporter is of the new type (can manipulate on internal data structure),
+	// use ExporterFactory.CreateTraceExporter.
+	return factory.CreateTraceExporter(ctx, creationParams, cfg)
 }
 
 // createMetricsExporter creates a metrics exporter based on provided factory type.
-func createMetricsExporter(factoryBase component.ExporterFactoryBase,
+func createMetricsExporter(
+	factory component.ExporterFactory,
 	logger *zap.Logger,
 	cfg configmodels.Exporter,
-) (component.MetricsExporterBase, error) {
-	if factory, ok := factoryBase.(component.ExporterFactory); ok {
-		creationParams := component.ExporterCreateParams{Logger: logger}
-		ctx := context.Background()
-
-		// If exporter is of the new type (can manipulate on internal data structure),
-		// use ExporterFactory.CreateMetricsExporter.
-		return factory.CreateMetricsExporter(ctx, creationParams, cfg)
-	}
+) (component.MetricsExporter, error) {
+	creationParams := component.ExporterCreateParams{Logger: logger}
+	ctx := context.Background()
 
-	// If exporter is of the old type (can manipulate on OC metrics only),
-	// use ExporterFactoryOld.CreateMetricsExporter.
-	return factoryBase.(component.ExporterFactoryOld).CreateMetricsExporter(logger, cfg)
+	// If exporter is of the new type (can manipulate on internal data structure),
```

Comment:
do we still need this comment?

Reply:
Will followup with removing of the ConsumerOld and will remove that as well.